logging/tests/unit/handlers/test_app_engine.py

```diff
@@ -24,34 +25,47 @@ def _get_target_class(self):
         return AppEngineHandler
 
     def _make_one(self, *args, **kw):
-        import tempfile
+        return self._get_target_class()(*args, **kw)
 
+    def test_ctor(self):
+        import os
         from google.cloud._testing import _Monkey
-        from google.cloud.logging.handlers import app_engine as _MUT
+        from google.cloud.logging.handlers.app_engine import _GAE_PROJECT_ENV, _GAE_SERVICE_ENV, _GAE_VERSION_ENV
 
-        tmpdir = tempfile.mktemp()
-        with _Monkey(_MUT, _LOG_PATH_TEMPLATE=tmpdir):
-            return self._get_target_class()(*args, **kw)
+        client = _Client(self.PROJECT)
+        with _Monkey(os, environ={_GAE_PROJECT_ENV: 'test_project',
+                                  _GAE_SERVICE_ENV: 'test_service',
+                                  _GAE_VERSION_ENV: 'test_version'}):
+            handler = self._make_one(client, transport=_Transport)
+        self.assertEqual(handler.client, client)
+        self.assertEqual(handler.resource.type, 'gae_app')
+        self.assertEqual(handler.resource.labels['project_id'], 'test_project')
+        self.assertEqual(handler.resource.labels['module_id'], 'test_service')
+        self.assertEqual(handler.resource.labels['version_id'], 'test_version')
 
-    def test_format(self):
-        import json
-        import logging
-
-        handler = self._make_one()
+    def test_emit(self):
+        client = _Client(self.PROJECT)
+        handler = self._make_one(client, transport=_Transport)
+        gae_resource = handler.get_gae_resource()
         logname = 'loggername'
         message = 'hello world'
-        record = logging.LogRecord(logname, logging.INFO, None,
-                                   None, message, None, None)
-        record.created = 5.03
-        expected_payload = {
-            'message': message,
-            'timestamp': {
-                'seconds': 5,
-                'nanos': int(.03 * 1e9),
-            },
-            'thread': record.thread,
-            'severity': record.levelname,
-        }
-        payload = handler.format(record)
-
-        self.assertEqual(payload, json.dumps(expected_payload))
+        record = logging.LogRecord(logname, logging, None, None, message,
+                                   None, None)
+        handler.emit(record)
+
+        self.assertEqual(handler.transport.send_called_with, (record, message, gae_resource))
+
+
+class _Client(object):
+
+    def __init__(self, project):
+        self.project = project
+
+
+class _Transport(object):
+
+    def __init__(self, client, name):
+        pass
```

Comment:
Yeah the reply got lost...but I have added the tests for the constructor args just now.